mondrian/src/it/java/mondrian/rolap/aggmatcher/AggSchemaScanTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * This software is subject to the terms of the Eclipse Public License v1.0
+ * Agreement, available at the following URL:
+ * http://www.eclipse.org/legal/epl-v10.html.
+ * You must accept the terms of that agreement to use this software.
+ *
+ * Copyright (c) 2002-2019 Hitachi Vantara..  All rights reserved.
+ */
+
+package mondrian.rolap.aggmatcher;
+
+import junit.framework.Assert;
+import mondrian.olap.Util;
+import mondrian.rolap.RolapConnection;
+import mondrian.rolap.RolapConnectionProperties;
+import mondrian.test.FoodMartTestCase;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+  * Test if AggSchemaScan and AggCatalogScan properties are used in JdbcSchema loadTablesOfType
+  *
+  */
+public class AggSchemaScanTest extends FoodMartTestCase {
+
+  public AggSchemaScanTest( String name ) {
+    super(name);
+  }
+
+
+  public void testAggScanPropertiesEmptySchema() throws Exception {
+
+    final RolapConnection rolapConn = (RolapConnection) getConnection();
+    final DataSource dataSource = rolapConn.getDataSource();
+    Connection sqlConnection = null;
+    try {
+      sqlConnection = dataSource.getConnection();
+      Util.PropertyList propertyList = new Util.PropertyList();
+      propertyList.put( RolapConnectionProperties.AggregateScanCatalog.name(), "bogus" );
+      propertyList.put( RolapConnectionProperties.AggregateScanSchema.name(), "bogus" );
+      JdbcSchema jdbcSchema = JdbcSchema.makeDB(dataSource);
+      jdbcSchema.resetAllTablesLoaded();
+      jdbcSchema.getTablesMap().clear();
+
+      jdbcSchema.loadTables( propertyList );
+      Assert.assertEquals( 0, jdbcSchema.getTablesMap().size() );
+    } finally {
+      if (sqlConnection != null) {
+        try {
+          sqlConnection.close();
+        } catch ( SQLException e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+
+  public void testAggScanPropertiesPopulatedSchema() throws Exception {
+
+    final RolapConnection rolapConn = (RolapConnection) getConnection();
+    final DataSource dataSource = rolapConn.getDataSource();
+    Connection sqlConnection = null;
+    try {
+      sqlConnection = dataSource.getConnection();
+      DatabaseMetaData dbmeta = sqlConnection.getMetaData();
+      if ( !dbmeta.supportsSchemasInTableDefinitions() && !dbmeta.supportsCatalogsInTableDefinitions() ) {
+        System.out.println( "Database does not support schema or catalog in table definitions.  Cannot run test." );
+        return;
+      }
+      Util.PropertyList propertyList = new Util.PropertyList();
+      boolean foundSchema = false;
+      // Different databases treat catalogs and schemas differently.  Figure out whether foodmart is a schema or catalog in this database
+      if ( dbmeta.supportsSchemasInTableDefinitions() ) {
```

Comment:
It might make sense to first just try the sqlConnection.getSchema() and sqlConnection.getCatalog() calls, since for many databases these will actually give something back.  You'd want to wrap the calls in a try/catch since some other jdbc clients will throw an exception.

This could allow you to avoid the hard assumption that the schema or catalog will be named 'foodmart' for the databases that support one or both of those getXXX methods.